File: android/app/src/test/java/com/xirigo/ecommerce/core/auth/FakeTokenStorage.kt
package com.xirigo.ecommerce.core.auth

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow

/**
 * In-memory [TokenStorage] implementation for use in unit tests.
 * Thread-safe via [MutableStateFlow].
 */
class FakeTokenStorage : TokenStorage {

    private val tokenFlow = MutableStateFlow<String?>(null)

    override suspend fun getAccessToken(): String? = tokenFlow.value

    override suspend fun saveAccessToken(token: String) {
        tokenFlow.value = token
    }

    override suspend fun clearTokens() {
        tokenFlow.value = null
    }

    override fun getAccessTokenFlow(): Flow<String?> = tokenFlow.asStateFlow()
}

File: ios/XiriGoEcommerceTests/Core/Auth/FakeTokenStorage.swift
import Foundation
@testable import XiriGoEcommerce

// MARK: - FakeTokenStorage

/// In-memory TokenStorage implementation for unit testing.
/// Avoids real Keychain access so tests run without device entitlements.
final class FakeTokenStorage: TokenStorage, @unchecked Sendable {
    // MARK: - Internal

    var storedToken: String?
    var saveCallCount = 0
    var clearCallCount = 0
    var getCallCount = 0

    // MARK: - TokenStorage

    func getAccessToken() -> String? {
        getCallCount += 1
        return storedToken
    }

    func saveAccessToken(_ token: String) {
        saveCallCount += 1
        storedToken = token
    }

    func clearTokens() {
        clearCallCount += 1
        storedToken = nil
    }
}
